### DISCUSSION TITLE
RetroPie experience

### DISCUSSION THREAD

**Discussion by StuartCameronCode** [Joshua-Riek/ubuntu-rockchip#103] [General]
I thought I'd share my overwhelmingly positive experience getting RetroPie installed and running on the Orange Pi 5 using this Ubuntu distribution.

# RetroPie installation

Installing the core RetroPie was basically seamless. So far I've tried Arcade/MAME, N64, PSP and with some custom tweaking I've managed to get the Dolphin emulator running with Wii & GameCube games.

EmulationStation has a blank bar at the top, its fullscreen position is offset from the top of the screen. I see there's a hack fix for this in the /usr/bin/emulationstation script, but it's not working for some reason. That's still on my TODO list.

# Performance

Wii and GameCube games when running at native resolution run at more or less full frame-rate. Increasing the resolution or enabling anti-aliasing drops the frame rate.

PSP runs at full frame rate at 2x resolution.

I can't get over how powerful this hardware is, and hopefully things will get even better as drivers improve.

# Wii / GameCube via Dolphin

## Startup Crash

Out of the box this emulator crashes on startup, you'll need to set the QT_QPA_PLATFORM value to `xcb`, I've pasted the line

```
export QT_QPA_PLATFORM=xcb
```
into my `/usr/bin/emulationstation` script.

## Shader crash

There's an issue listed here https://bugs.dolphin-emu.org/issues/13198 affecting the Orange PI 5. You'll need to check out https://github.com/dolphin-emu/dolphin/pull/11628 and build Dolphin locally with this fix until this PR gets merged. I build a "portable" version so everything's standalone.

## GFX.ini

You'll need to set Dolphin to use OpenGL ES.

In your freshly build `dolphin` folder edit `dolphin/build/Binaries/user/Config/GFX.ini` and ensure you've got this:

```
[Settings]
PreferGLES = True
```

## Config

You'll want to launch Dolphin manually first-time around to configure it to launch in full-screen and set up your controller mappings.

```
QT_QPA_PLATFORM=xcb ./dolphin-emu
```

## Setup Custom Dolphin for EmulationStation

Add a line to `/opt/retropie/configs/wii/emulators.xml` and `/opt/retropie/configs/gc/emulators.xml` such as:
`dolphin-gui-opi="/home/[username]/dolphin/build/Binaries/dolphin-emu -b -e %ROM%"`

Then launch a game image, press a button from the runner screen and select the new `dolphin-gui-opi` emulator config.

# Driver Support

Everything I've tried from a Wifi dongle to PS4 controller and my zero delay USB encoders for my arcade cabinet have all just worked flawlessly.

**Comment by StuartCameronCode** [comment]
I would like to add that @Joshua-Riek has done an AMAZING job with this image, it's incredibly stable and "just works". THANK YOU!!

**Comment by HSTe** [comment]
I guess this is the same problem with screen:
https://retropie.org.uk/forum/topic/33999/emulationstation-loads-slightly-too-low-ubuntu-22-04/4

Have you tested different video drivers?

**Reply by StuartCameronCode** [reply]
I worked out about 30 mins ago that commenting out the full screen hack and the setting of SDL to Wayland in /use/bin/emulationstation fixes the issue, without having to change anything global.

**Comment by HectorPulido** [comment]
Did you try it with the wiimotes? Mine doesn't work 😢
Only works for a second if I run dolphin as sudo and then gets disconnected

**Reply by StuartCameronCode** [reply]
Nah, I haven't gone down that route, I was mostly after GameCube emulation.

**Reply by diegov-au** [reply]
My wiimotes work great with a Dolphin Bar. Ensure continuous scanning is switched on under controller settings.
